conjure-java-core/src/main/java/com/palantir/conjure/java/types/ErrorGenerator.java

```diff
@@ -153,21 +165,164 @@ private JavaFile generateErrorTypesForNamespace(
+        }
 
         return JavaFile.builder(conjurePackage, typeBuilder.build())
                 .skipJavaLangImports(true)
                 .indent("    ")
                 .build();
     }
 
+    private static List<TypeSpec> generateRemoteExceptionTypes(
+            ClassName errorsClassName, List<ErrorDefinition> errorDefinitions) {
+        return errorDefinitions.stream()
+                .map(def -> generateRemoteExceptionType(errorsClassName, def))
+                .toList();
+    }
+
+    private static TypeSpec generateRemoteExceptionType(ClassName errorsClassName, ErrorDefinition errorDefinition) {
+        String remoteExceptionClassName = errorDefinition.getErrorName().getName() + "Exception";
+        ClassName serializableErrorClassName =
+                errorsClassName.nestedClass(errorDefinition.getErrorName().getName() + "SerializableError");
+
+        TypeSpec.Builder classBuilder = TypeSpec.classBuilder(remoteExceptionClassName)
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL)
+                .superclass(RemoteException.class)
+                .addSuperinterface(ParameterizedTypeName.get(
+                        ClassName.get(SerializableErrorProvider.class),
+                        errorsClassName.nestedClass(ErrorGenerationUtils.errorParametersClassName(
+                                errorDefinition.getErrorName().getName()))))
+                .addField(FieldSpec.builder(serializableErrorClassName, "error")
+                        .addModifiers(Modifier.PRIVATE)
+                        .build())
+                .addMethod(MethodSpec.constructorBuilder()
+                        .addModifiers(Modifier.PUBLIC)
+                        .addParameter(serializableErrorClassName, "error")
+                        .addParameter(int.class, "status")
+                        .addStatement("super(error.toSerializableError(), status)")
+                        .addStatement("this.error = error")
+                        .build())
+                .addMethod(MethodSpec.methodBuilder("error")
+                        .addModifiers(Modifier.PUBLIC)
+                        .returns(serializableErrorClassName)
+                        .addStatement("return error")
+                        .build());
+
+        return classBuilder.build();
+    }
+
+    private static List<TypeSpec> generateSerializableErrors(List<ErrorDefinition> errorDefinitions) {
+        return errorDefinitions.stream()
+                .map(ErrorGenerator::generateSerializableError)
+                .toList();
+    }
+
+    private static TypeSpec generateSerializableError(ErrorDefinition errorDefinition) {
+        String serializableErrorClassName = errorDefinition.getErrorName().getName() + "SerializableError";
+        String parameterClassNameString = ErrorGenerationUtils.errorParametersClassName(
+                errorDefinition.getErrorName().getName());
```

Comment:
vnit: could streamline these lines a bit by making `ErrorGenerationUtils.errorParametersClassName` take an `ErrorDefinition` (unless it's not always the same field that we're getting?)